src/libraries/System.Text.Json/tests/Serialization/StreamTests.cs

```diff
@@ -0,0 +1,368 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+
+using Xunit;
+
+namespace System.Text.Json.Serialization.Tests
+{
+    public static partial class StreamTests
+    {
+        // To hit all possible continuation positions inside the tested object,
+        // the outer-class padding needs to be between 5 and 116 bytes long.
+
+        // {"S":"xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx","C":{"A":null,"B":"Hello","C":42,"D":null,"E":3.14E+17,"F":null,"G":true,"H":null,"I":[42,17],"J":{"A":null,"B":7}}}
+        // |<------------------------------------------------128 byte buffer------------------------------------------------------------->|
+        // {"S":"xxxxx","C":{"A":"Hello","B":null,"C":42,"D":null,"E":3.14E+17,"F":null,"G":true,"H":null,"I":[42,17],"J":{"A":null,"B":7}}}
+
+        private const int MinPaddingLength = 5;
+        private const int MaxPaddingLength = 116;
+
+        private static IEnumerable<int> ContinuationPaddingLengths
+            => Enumerable.Range(MinPaddingLength, MaxPaddingLength - MinPaddingLength + 1);
+
+        private static IEnumerable<bool> IgnoreNullValues
+            => new[] { true, false };
```

Comment:
These are great tests and a clever way to get a lot more continuation coverage! Nicely done.

If you are up for it, I think it would be really valuable to do a similar thing with the `Utf8JsonReader` overload of `JsonSerialier.Deserialize` (possibly in a separate PR), which also supports continuation/partial buffers via the `ReadOnlySequence<byte>` (where the whole JSON is available in memory, but split in non-contiguous chunks).
https://docs.microsoft.com/en-us/dotnet/api/system.text.json.jsonserializer.deserialize?view=netcore-3.1#System_Text_Json_JsonSerializer_Deserialize_System_Text_Json_Utf8JsonReader__System_Type_System_Text_Json_JsonSerializerOptions_
https://docs.microsoft.com/en-us/dotnet/api/system.text.json.utf8jsonreader.-ctor?view=netcore-3.1#System_Text_Json_Utf8JsonReader__ctor_System_Buffers_ReadOnlySequence_System_Byte__System_Boolean_System_Text_Json_JsonReaderState_

Reply:
Thanks!
I will have a look.